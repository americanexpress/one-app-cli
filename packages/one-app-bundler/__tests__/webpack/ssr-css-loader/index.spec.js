/*
 * Copyright 2019 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations
 * under the License.
 */

const SSRCSSLoader = require('../../../webpack/loaders/ssr-css-loader');

jest.mock('path', () => ({
  resolve: (dir, filename) => `/path/to/one-app-bundler/webpack/ssr-css-loader/${filename}`,
}));

describe('SSR CSS loader', () => {
  const consoleSpy = jest.spyOn(console, 'log');

  beforeEach(() => consoleSpy.mockClear());

  it('should throw an error when the css-loader is not present', () => {
    expect(() => SSRCSSLoader('no css-loader here!')).toThrowErrorMatchingSnapshot();
    expect(() => SSRCSSLoader(`var ___CSS_LOADER_API_IMPORT___ = __webpack_require__(/*! css-loader/dist/runtime/api.js */ "css-loader/dist/runtime/api.js");
    exports = ___CSS_LOADER_API_IMPORT___();`)).toThrowErrorMatchingSnapshot();
    expect(() => SSRCSSLoader(`var ___CSS_LOADER_API_IMPORT___ = __webpack_require__(/*! ../../css-loader/dist/runtime/api.js */ "../../css-loader/dist/runtime/api.js");
    exports = ___CSS_LOADER_API_IMPORT___(true);`)).toThrowErrorMatchingSnapshot();
    expect(() => SSRCSSLoader(`
    // Imports
    var ___CSS_LOADER_API_IMPORT___ = __webpack_require__(/*! ../../see-ess-ess-loader/lib/see-ess-ess-base.js */ "../../see-ess-ess-loader/lib/see-ess-ess-base.js");
    exports = ___CSS_LOADER_API_IMPORT___(undefined);
    // Module
    exports.push([module.i, ".my-root__ErrorLayout__ErrorLayout___2B91F {\n  margin: 0 auto;\n}\n\n.my-root__ErrorLayout__main___2IftH {\n  max-width: 1000px;\n  width: 100%;\n  margin: 0 auto;\n  padding: 10px;\n}", ""]);
    // Exports
    exports.locals = {
    \t"ErrorLayout": "my-root__ErrorLayout__ErrorLayout___2B91F",
    \t"main": "my-root__ErrorLayout__main___2IftH"
    };
    module.exports = exports;
    `)).toThrowErrorMatchingSnapshot();
  });

  it('should remove css-loader\'s css-base, replace its push and export the locals', () => {
    expect(SSRCSSLoader(`var ___CSS_LOADER_API_IMPORT___ = __webpack_require__("../../node_modules/css-loader/dist/runtime/api.js");
    exports = ___CSS_LOADER_API_IMPORT___(undefined);`)).toMatchSnapshot();
    expect(SSRCSSLoader(`var ___CSS_LOADER_API_IMPORT___ = __webpack_require__("../../css-loader/dist/runtime/api.js");
    exports = ___CSS_LOADER_API_IMPORT___();`)).toMatchSnapshot();
    expect(SSRCSSLoader(`var ___CSS_LOADER_API_IMPORT___ = __webpack_require__("./css-loader/dist/runtime/api.js");
    exports = ___CSS_LOADER_API_IMPORT___();`)).toMatchSnapshot();
    expect(SSRCSSLoader(`var ___CSS_LOADER_API_IMPORT___ = __webpack_require__("./css-loader/dist/runtime/api.js");
    exports = ___CSS_LOADER_API_IMPORT___(undefined);`)).toMatchSnapshot();
    expect(SSRCSSLoader(`
      // Imports
      var ___CSS_LOADER_API_IMPORT___ = __webpack_require__(/*! ./node_modules/css-loader/dist/runtime/api.js */ "./node_modules/css-loader/dist/runtime/api.js");
      exports = ___CSS_LOADER_API_IMPORT___(undefined);
      // Module
      exports.push([module.i, ".my-root__ErrorLayout__ErrorLayout___2B91F {\n  margin: 0 auto;\n}\n\n.my-root__ErrorLayout__main___2IftH {\n  max-width: 1000px;\n  width: 100%;\n  margin: 0 auto;\n  padding: 10px;\n}", ""]);
      // Exports
      exports.locals = {
      \t"ErrorLayout": "my-root__ErrorLayout__ErrorLayout___2B91F",
      \t"main": "my-root__ErrorLayout__main___2IftH"
      };
      module.exports = exports;
    `)).toMatchSnapshot();
    expect(SSRCSSLoader(`var ___CSS_LOADER_API_IMPORT___ = __webpack_require__("../../node_modules/css-loader/dist/runtime/api.js");
    exports = ___CSS_LOADER_API_IMPORT___(false);`)).toMatchSnapshot();
    expect(SSRCSSLoader(`var ___CSS_LOADER_API_IMPORT___ = __webpack_require__("./css-loader/dist/runtime/api.js");
    exports = ___CSS_LOADER_API_IMPORT___(false);`)).toMatchSnapshot();
    expect(SSRCSSLoader(`
      // Imports
      var ___CSS_LOADER_API_IMPORT___ = __webpack_require__(/*! ../../css-loader/dist/runtime/api.js */ "../../css-loader/dist/runtime/api.js");
      exports = ___CSS_LOADER_API_IMPORT___(false);
      // Module
      exports.push([module.i, ".my-root__ErrorLayout__ErrorLayout___2B91F {\n  margin: 0 auto;\n}\n\n.my-root__ErrorLayout__main___2IftH {\n  max-width: 1000px;\n  width: 100%;\n  margin: 0 auto;\n  padding: 10px;\n}", ""]);
      // Exports
      exports.locals = {
      \t"ErrorLayout": "my-root__ErrorLayout__ErrorLayout___2B91F",
      \t"main": "my-root__ErrorLayout__main___2IftH"
      };
      module.exports = exports;
    `)).toMatchSnapshot();
  });
});
