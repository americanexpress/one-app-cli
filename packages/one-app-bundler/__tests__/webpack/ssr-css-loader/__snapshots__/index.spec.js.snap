// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 1`] = `""`;

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 2`] = `""`;

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 3`] = `""`;

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 4`] = `""`;

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 5`] = `
"
      // Imports
      
      // Module
      require(\\"/path/to/one-app-bundler/webpack/ssr-css-loader/css-base.js\\")().push([module.i, \\".my-root__ErrorLayout__ErrorLayout___2B91F {
  margin: 0 auto;
}

.my-root__ErrorLayout__main___2IftH {
  max-width: 1000px;
  width: 100%;
  margin: 0 auto;
  padding: 10px;
}\\", \\"\\"]);
      // Exports
      exports = module.exports = {
      	\\"ErrorLayout\\": \\"my-root__ErrorLayout__ErrorLayout___2B91F\\",
      	\\"main\\": \\"my-root__ErrorLayout__main___2IftH\\"
      };
      module.exports = exports;
    "
`;

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 6`] = `""`;

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 7`] = `""`;

exports[`SSR CSS loader should remove css-loader's css-base, replace its push and export the locals 8`] = `
"
      // Imports
      
      // Module
      require(\\"/path/to/one-app-bundler/webpack/ssr-css-loader/css-base.js\\")().push([module.i, \\".my-root__ErrorLayout__ErrorLayout___2B91F {
  margin: 0 auto;
}

.my-root__ErrorLayout__main___2IftH {
  max-width: 1000px;
  width: 100%;
  margin: 0 auto;
  padding: 10px;
}\\", \\"\\"]);
      // Exports
      exports = module.exports = {
      	\\"ErrorLayout\\": \\"my-root__ErrorLayout__ErrorLayout___2B91F\\",
      	\\"main\\": \\"my-root__ErrorLayout__main___2IftH\\"
      };
      module.exports = exports;
    "
`;

exports[`SSR CSS loader should throw an error when the css-loader is not present 1`] = `
"could not find the css-loader in
no css-loader here!"
`;

exports[`SSR CSS loader should throw an error when the css-loader is not present 2`] = `
"could not find the css-loader in
var ___CSS_LOADER_API_IMPORT___ = __webpack_require__(/*! css-loader/dist/runtime/api.js */ \\"css-loader/dist/runtime/api.js\\");
    exports = ___CSS_LOADER_API_IMPORT___();"
`;

exports[`SSR CSS loader should throw an error when the css-loader is not present 3`] = `
"could not find the css-loader in
var ___CSS_LOADER_API_IMPORT___ = __webpack_require__(/*! ../../css-loader/dist/runtime/api.js */ \\"../../css-loader/dist/runtime/api.js\\");
    exports = ___CSS_LOADER_API_IMPORT___(true);"
`;

exports[`SSR CSS loader should throw an error when the css-loader is not present 4`] = `
"could not find the css-loader in

    // Imports
    var ___CSS_LOADER_API_IMPORT___ = __webpack_require__(/*! ../../see-ess-ess-loader/lib/see-ess-ess-base.js */ \\"../../see-ess-ess-loader/lib/see-ess-ess-base.js\\");
    exports = ___CSS_LOADER_API_IMPORT___(undefined);
    // Module
    exports.push([module.i, \\".my-root__ErrorLayout__ErrorLayout___2B91F {
  margin: 0 auto;
}

.my-root__ErrorLayout__main___2IftH {
  max-width: 1000px;
  width: 100%;
  margin: 0 auto;
  padding: 10px;
}\\", \\"\\"]);
    // Exports
    exports.locals = {
    	\\"ErrorLayout\\": \\"my-root__ErrorLayout__ErrorLayout___2B91F\\",
    	\\"main\\": \\"my-root__ErrorLayout__main___2IftH\\"
    };
    module.exports = exports;
    "
`;
